ui/store/actions.js

```diff
@@ -673,6 +673,49 @@ const updateMetamaskStateFromBackground = () => {
   });
 };
 
+export function updateSwapApprovalTransaction(txId, txSwapApproval) {
+  return async (dispatch) => {
+    try {
+      await promisifiedBackground.updateSwapApprovalTransaction(
+        txId,
+        txSwapApproval,
+      );
+    } catch (error) {
+      dispatch(txError(error));
+      dispatch(goHome());
+      log.error(error.message);
+      throw error;
+    }
+
+    dispatch(
+      updateTransactionParams(txSwapApproval.id, txSwapApproval.txParams),
+    );
+    const newState = await updateMetamaskStateFromBackground();
+    dispatch(updateMetamaskState(newState));
+    dispatch(showConfTxPage({ id: txSwapApproval.id }));
```

Comment:
I don't think `dispatch(showConfTxPage({ id: txSwapApproval.id }));` is needed any more

Reply:
Yeah, fixed [here](https://github.com/MetaMask/metamask-extension/pull/13641/commits/b9c21cb1ef00be88bf696fa3e0d82c5c935bcf1f)